tools/fapi/tss2_template.c

```diff
@@ -563,7 +567,13 @@ int main(int argc, char *argv[]) {
      * Cleanup contexts & memory allocated for the modified argument vector
      * passed to execute_tool.
      */
-    Fapi_Finalize (&fctx);
+    if (strcmp(tool_name, "gettpm2object")) {
+        /*
+         * Fapi_Finalize was already executed by gettpm2object.
+         */
+        Fapi_Finalize (&fctx);
```

Comment:
Why do we have to change the flow for this tool? I would imagine it could still call finalize in the framework flow like normal.

Reply:
> Why do we have to change the flow for this tool? I would imagine it could still call finalize in the framework flow like normal.

The tool needed an esys context. Unfortunately it's not possible to get the esys context from the fapi context via fapi API.
With the socket interface I had problems to create the esys context. Therefore the fapi context was closed before calling Esys_Initialize.

Reply:
Sorry I forgot that we have Fapi_GetTcti, which can be used to create an esys context. I will change it.